### DISCUSSION TITLE
Image click detection

### DISCUSSION THREAD

**Discussion by Zephyriyn** [zauberzeug/nicegui#1985] [Q&A]
### Description

how do i check which image was clicked out of the images in the loop via .on('click') in a scenario such as this: 
```py
    for result in results:
        with search_results:
            with ui.image("https://image.tmdb.org/t/p/w500"+result['poster_path']).style("width:400px;height:100px").on('clck'): #??
                ui.label(result['title'])
```

**Comment by nibinho** [comment]
I don't know how your `results` or your `search_results` look like, but here is how you can access the click event on an image in a loop:

```python
from nicegui import events, ui

results = [
    {"path": "350", "title": "350"},
    {"path": "360", "title": "360"},
    {"path": "370", "title": "370"},
    {"path": "380", "title": "380"},
]


def handle_click(e: events.ClickEventArguments):
    ui.notify(e.sender._props)


with ui.column():
    for result in results:
        with ui.image(f"https://picsum.photos/id/{result['path']}/640/360").style(
            "width:400px;height:100px"
        ).props(f"default={result['title']}").on("click", handle_click):
            ui.label(result["title"])

ui.run()
```

The `default` prop is taken from the [QImg documentation](https://quasar.dev/vue-components/img) (the prop has no effect in Nicegui at the moment and could be named differently), `e.sender._props` gets you all props that were set on the element. 
If there is any other, more official feeling way to do this I am not aware of it.
